test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/DataAnnotationsModelValidatorTest.cs

```diff
@@ -220,55 +220,6 @@ public void ValidateReturnsMemberNameIfItIsDifferentFromDisplayName()
 #endif
 
         [Fact]
-        public void GetClientValidationRules_ReturnsEmptyRuleSet()
```

Comment:
>where is GetErrorMessage() checked?
The individual adapters eg look at `StringLengthAttributeAdapterTest`. 
I do not see much value in testing this for the abstract base class. Dont feel strongly against it though. If you do feel strongly for it. I will add a new one.  (Please note the test I have removed had nothing to do with GetErrorMessage.

> what confirms that the [data annotations] client validator provider successfully returns an empty rule set for an unknown ValidationAttribute

It doesnt return an empty rule set. There is no client validator added in such a scenario.

>what confirms that the other provider handles the IClientModelValidator case in general (not just for [Remote])?

look at `DefaultModelValidatorProviderTest`.